tests/unittests/CoreGraphics.drawing/ImageHelpers.cpp

```diff
@@ -0,0 +1,114 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include "ImageHelpers.h"
+
+#include <Starboard/SmartTypes.h>
+#include <ImageIO/ImageIO.h>
+
+#include <memory>
+
+#include <windows.h>
+
+CGImageRef _CGImageCreateFromPNGFile(CFStringRef filename) {
+    woc::unique_cf<CGDataProviderRef> dataProvider{ CGDataProviderCreateWithFilename(
+        CFStringGetCStringPtr(filename, kCFStringEncodingUTF8)) };
+    if (!dataProvider) {
+        return nullptr;
+    }
+
+    woc::unique_cf<CGImageRef> image{ CGImageCreateWithPNGDataProvider(dataProvider.get(), nullptr, FALSE, kCGRenderingIntentDefault) };
+    return image.release();
+}
+
+CFDataRef _CFDataCreatePNGFromCGImage(CGImageRef image) {
+    // Estimate the image data size to be as large as its raw pixel buffer.
+    // This will never be hit; but if it does, CFData will grow intelligently regardless.
+    size_t sizeEstimate = CGImageGetHeight(image) * CGImageGetBytesPerRow(image);
+    woc::unique_cf<CFMutableDataRef> imageData{ CFDataCreateMutable(nullptr, sizeEstimate) };
+    if (!imageData) {
+        return nullptr;
+    }
+
+    woc::unique_cf<CGImageDestinationRef> imageDest{ CGImageDestinationCreateWithData(imageData.get(), CFSTR("public.png"), 1, nullptr) };
+    if (!imageDest) {
+        return nullptr;
+    }
+
+    CGImageDestinationAddImage(imageDest.get(), image, nullptr);
+    CGImageDestinationFinalize(imageDest.get());
+
+    return imageData.release();
+}
+
+bool _WriteCFDataToFile(CFDataRef data, CFStringRef filename) {
+    std::unique_ptr<char[]> owningFilenamePtr;
+
+    char* rawFilename = const_cast<char*>(CFStringGetCStringPtr(filename, kCFStringEncodingUTF8));
```

Comment:
I know that this function has given us some trouble in CT land, does this work consistently enough that it's worth trying?